winsup/cygwin/path.cc

```diff
@@ -2154,6 +2158,12 @@ symlink_worker (const char *oldpath, path_conv &win32_newpath, bool isdevice)
 	      __seterrno ();
 	      __leave;
 	    }
+	  /* With deepcopy fall back? Let's do that, then */
+	  if (res == -1 && wsym_type == WSYM_native_or_deepcopy)
```

Comment:
Note that if target `isspecial` res will be `-2`, so will still fall back to default.  This is probably reasonable, since deepcopy would just fall back to sysfile in that case.  default will let it try wsl before falling back to sysfile.  I'm thinking it might be best to make deepcopy fall back to default instead of sysfile anyway.

Reply:
making deepcopy fall back to wsl instead of sysfile breaks one git test which tries to commit a symlink to /dev/null.  The result of doing so is almost certainly not helpful (I bet it commits the sysfile, not a symlink), but when it's a wsl symlink it fails with `Function not implemented`

Reply:
> when it's a wsl symlink it fails with `Function not implemented`

That's my fault, as I have not managed to spend time on upstreaming Git for Windows' symlink support: https://github.com/git/git/compare/master...dscho:git:support-symlinks-on-windows

Reply:
I had a quick look at those changes, and I didn't see any handling of wsl symlinks in there, so I don't think that would help.

Reply:
Huh. I misremembered. A `git grep IO_REPARSE_TAG_LX_SYMLINK` indeed comes up empty...